File: web/src/services/linux-user.ts
import request from '../utils/request'
import type { PageResult } from './audit'

export interface LinuxUserItem {
  id: number
  username: string
  asset_id: number
  hostname: string
  ip: string
  uid: number
  gid: number
  home: string
  shell: string
  sudo: boolean
  status: string
}

export interface CreateLinuxUserParams {
  username: string
  asset_ids: number[]
  shell?: string
  sudo?: boolean
}

export function getLinuxUsers(params?: { username?: string; page?: number; page_size?: number }): Promise<PageResult<LinuxUserItem>> {
  return request.get('/linux-users', { params })
}

export function createLinuxUser(data: CreateLinuxUserParams): Promise<void> {
  return request.post('/linux-users', data)
}

export function deleteLinuxUser(username: string, assetIds: number[]): Promise<void> {
  return request.delete(`/linux-users/${username}`, { data: { asset_ids: assetIds } })
}

File: internal/dto/linux_user.go
package dto

// CreateLinuxUserRequest 创建 Linux 用户请求
type CreateLinuxUserRequest struct {
	Username string `json:"username" binding:"required,max=64"`
	AssetIDs []uint `json:"asset_ids" binding:"required"`
	Shell    string `json:"shell"`
	Sudo     bool   `json:"sudo"`
}

// DeleteLinuxUserRequest 删除 Linux 用户请求
type DeleteLinuxUserRequest struct {
	AssetIDs []uint `json:"asset_ids" binding:"required"`
}

// LinuxUserResponse Linux 用户响应
type LinuxUserResponse struct {
	ID       uint   `json:"id"`
	Username string `json:"username"`
	AssetID  uint   `json:"asset_id"`
	Hostname string `json:"hostname"`
	IP       string `json:"ip"`
	UID      int    `json:"uid"`
	GID      int    `json:"gid"`
	Home     string `json:"home"`
	Shell    string `json:"shell"`
	Sudo     bool   `json:"sudo"`
	Status   string `json:"status"`
}
